holmes.go

```diff
@@ -103,6 +105,7 @@ func (h *Holmes) DisableMemDump() *Holmes {
 func (h *Holmes) Start() {
 	atomic.StoreInt64(&h.stopped, 0)
 	h.initEnvironment()
+	go h.startMemCronJob()
```

Comment:
it's a common scenario and should support memory, goroutine and cpu etc.
`h.startCronJob()` ?

Reply:
> it's a common scenario and should support memory, goroutine and cpu etc. `h.startCronJob()` ?
For the warning scenario, the memory cron job is a useful feature, as I said at [this issue](https://github.com/mosn/holmes/issues/42).

If we need stats and analyze the performance of the application, we can add cpu/goroutine cron job feature.
And, in my plan, we can merge those profiles that got by corn job as a big profile, then analyze this big profile, we can get some information such like hot-spot code and slowest function in our application. 

But I am not sure the feature is holmes need? Because holmes is a **self_dump** library, its main scenario is fast finding the `bug` at warning time instead of a performance analyzing tool or platform. 

ps: for "a performance analyzing tool or platform", this thing is what I am thinking recently, I am glad to contact any idea with you.

Reply:
Can you add me on WeChat?  I have emailed you.

Reply:
> > it's a common scenario and should support memory, goroutine and cpu etc. `h.startCronJob()` ?
> 
> For the warning scenario, the memory cron job is a useful feature, as I said at [this issue](https://github.com/mosn/holmes/issues/42).
> 
> If we need stats and analyze the performance of the application, we can add cpu/goroutine cron job feature. And, in my plan, we can merge those profiles that got by corn job as a big profile, then analyze this big profile, we can get some information such like hot-spot code and slowest function in our application.
> 
> But I am not sure the feature is holmes need? Because holmes is a **self-aware** library, its main scenario is fast finding the `bug` at warning time(I got it from readme.md) instead of a performance analyzing tool or platform.
> 
> ps: for "a performance analyzing tool or platform", this thing is what I am thinking recently, I am glad to contact any idea with you.

This is the original definition of continuous profiling, 
but previously we run holmes in a large scale scene,
keep dumping at high frequency will affect the performance of app,
slow memory leaks were detected by setting the third param to each metrics(memory, goroutines)

Reply:
> This is the original definition of continuous profiling, previously we run holmes in a large scale scene, keep dumping at high frequency would bring negative effect to the performance of app, slow memory leaks were detected by setting the third param to each metrics(memory, goroutines)
> 
> ```go
>     holmes.WithMemDump(30, 25, 80/*slow leaks will also take your app to 80% memory usage*/),
> 
>     holmes.WithGoroutineDump(500, 25, 20000/*slowly leaks until we reach 20k goroutine*/),
> ```

Emmm, you make a good point, It seems that we don't need this cron feature. Unless the application OOM during cooldown period. 
And I am not sure about this cron job feature we still need?  @doujiang24

Reply:
> And I am not sure about this cron job feature we still need?

@Jun10ng I do think it's still worth adding this feature. It may help us to find bugs earlier in another way. And it's not enabled by default, it looks good enough for me.

Reply:
I think this cron job could be a general feature, one scenario that we can collect the baseline of the app about mem,cpu and gourtine per hour.  as @Jun10ng said: "a performance analyzing tool or platform"